packages/contracts/src/chugsplash/deploy.ts

```diff
@@ -0,0 +1,45 @@
+/* Imports: External */
+import { injectL2Context } from '@eth-optimism/core-utils'
+import hre from 'hardhat'
+import { ethers } from 'ethers'
+
+/* Imports: Internal */
+import { getContractFactory } from '..'
+import { makeActionBundleFromConfig } from './hardhat-tools'
+
+const DEPLOYER_PRIVATE_KEY = process.env.DEPLOYER_PRIVATE_KEY
+const L2_NODE_URL = process.env.L2_NODE_URL
+const CHUGSPLASH_DEPLOYER_ADDRESS = process.env.CHUGSPLASH_DEPLOYER_ADDRESS
+const UPGRADE_CONFIG_PATH = process.env.UPGRADE_CONFIG_PATH
+
+const deploy = async () => {
+  const l2Provider = injectL2Context(
+    new ethers.providers.JsonRpcProvider(L2_NODE_URL)
+  )
+  const signer = new ethers.Wallet(DEPLOYER_PRIVATE_KEY, l2Provider)
+
+  const config = require(UPGRADE_CONFIG_PATH)
+  const actionBundle = await makeActionBundleFromConfig(hre, config)
+
+  const deployerContract = getContractFactory(
+    'L2ChugSplashDeployer',
+    signer
+  ).attach(CHUGSPLASH_DEPLOYER_ADDRESS)
+
+  while ((await deployerContract.currentBundleHash()) !== actionBundle.root) {
+    // Sleep 1 second
+    await new Promise((r) => setTimeout(r, 1_000))
+  }
+
+  const approveTx = await deployerContract.approveTransactionBundle(
+    actionBundle.root,
+    actionBundle.actions.length
+  )
```

Comment:
called this function and assumed these are the inputs

https://github.com/ethereum-optimism/optimism/blob/39d943765eceac034528a9d7dd168ff65d628056/packages/contracts/contracts/chugsplash/L2ChugSplashDeployer.sol#L141-L144

Reply:
Ah we shouldn't need to call `approveTransactionBundle` here. The assumption is that `approveTransactionBundle` will be called by something (or someone) else, e.g., a multisig wallet. Instead we want to execute a loop that looks something like:

```ts
for (const action of actionBundle.actions) {
  await deployerContract.executeAction(
    action.action,
    action.proof
  )
}
```

(plus some error handling)